recipe/meta.yaml

```diff
@@ -1,61 +1,44 @@
-{% set name = "onnx" %}
 {% set version = "1.15.0" %}
 
 package:
-  name: {{ name|lower }}
+  name: onnx-split
   version: {{ version }}
 
 source:
   url: https://github.com/onnx/onnx/archive/v{{ version }}.tar.gz
   sha256: c757132e018dd0dd171499ef74fca88b74c5430a20781ec53da19eb7f937ef68
   patches:
-    - 0001-Link-to-abseil_dll.patch  # [win and (libprotobuf != "3.21")]
+    - patches/0001-Link-to-abseil_dll.patch
+    - patches/0002-Let-conda-forge-define-CXX-standard.patch
 
 build:
-  number: 3
-  entry_points:
-    - check-model = onnx.bin.checker:check_model
-    - check-node = onnx.bin.checker:check_node
-    - backend-test-tools = onnx.backend.test.cmd_tools:main
+  number: 4
 
-requirements:
-  build:
-    - python                                 # [build_platform != target_platform]
-    - cross-python_{{ target_platform }}     # [build_platform != target_platform]
-    - numpy                                  # [build_platform != target_platform]
-    - pybind11                               # [build_platform != target_platform]
-    - {{ compiler('c') }}
-    - {{ compiler('cxx') }}
-    - cmake
-    - make
-    - libprotobuf
-  host:
-    - python
-    - pip
-    - protobuf
-    - libprotobuf
-    - pytest-runner
-    - ninja
-    - pybind11
-    - numpy
-  run:
-    - python
-    - protobuf
-    - {{ pin_compatible('numpy') }}
-    - typing-extensions >=3.6.2.1
-
-test:
-  requires:
-    - pip
-  imports:
-    - onnx
-  commands:
-    - pip check
-    - check-model --help
-    - check-node --help
-    - backend-test-tools --help
-    - test -f "$PREFIX/lib/libonnx${SHLIB_EXT}"  # [unix]
-    - test -f "$PREFIX/lib/cmake/ONNX/ONNXConfig.cmake"  # [unix]
+outputs:
+  - name: libonnx
+    script: build_libonnx.sh  # [unix]
+    script: bld_libonnx.bat   # [win]
+    build:
+      run_exports:
+        # No stable C API offered for onnx
+        - {{ pin_subpackage('libonnx', max_pin='x.x.x') }}
```

Comment:
@cbourjau this pin would ensure that the runtime version matches the compilation-time version down to the patch number.

I'm hoping that libonnx can be used for downstream projects to avoid them rebuilding everything (and all dependencies)

My goal is eventually to build out tensorrt
https://github.com/onnx/onnx-tensorrt

Reply:
Is this safe to do, though? ONNX does not provide a C-API so any linking to a dynamic library would depend on an unstable ABI.

Reply:
As long as we're building for precisely the same release, the ABI should not change. The ABI should only change between versions not depending on build flags.

Reply:
Please correct me if I'm wrong, but I was under the impression that compiler flags may change the ABI. This is also how I understand this section of the gcc documentation: https://gcc.gnu.org/onlinedocs/libstdc++/manual/abi.html

Wouldn't this mean that a package linking to libonnx would have to follow implicit constraints on the compiler and the flags in use?

Reply:
We have a common set of compiler flags we use across all of conda-forge. The flags you are referring to should not be used as they will break the compatibility. If you break compatibility because you change the ABI using this flags, you will have to invest a lot of extra work. We should add complicated logic because of this very very rare case.

This is now drifting into a fundamental discussion about dynamic linkage across conda-forge. We are using what @hmaarrfk is doing in all of conda-forge (depending on the ABI guarantees between releases).